### DISCUSSION TITLE
mpv 的glsl shader执行流程

### DISCUSSION THREAD

**Discussion by feixiaku** [hooke007/mpv_PlayKit#33] [Q&A 问答]
https://github.com/hooke007/MPV_lazy/blob/master/shaders/FSRCNNX_x2_16-0-4-1_anime_enhance.glsl
您好，请教两个问题：
1. 在这个glsl里面，存在多个hook，他们是按照顺序执行的吗？
2.  hook函数前面的这些注释是什么意思？多个//!BIND，以及//!HOOK(看mpv的文档说明，没看明白) 比如：
3. //!HOOK LUMA//!WHEN OUTPUT.w LUMA.w / 1.300 &gt; OUTPUT.h LUMA.h / 1.300 &gt; *//!DESC (FSRCNNX_x2_16-0-4-1_a_e) mapping 1_1//!BIND FEATURE1//!BIND FEATURE2//!BIND FEATURE3//!BIND FEATURE4//!SAVE MODEL21//!COMPONENTS 4

多谢~

**Comment by hooke007** [comment]
>在这个glsl里面，存在多个hook，他们是按照顺序执行的吗？

是的依次执行，同样也可以通过stats的信息验证。
![Snipaste_2021-05-18_13-42-44](https://user-images.githubusercontent.com/41094733/118597201-f40c4680-b7de-11eb-8c21-58c39c91a043.jpg)


>hook函数前面的这些注释是什么意思？

面向用户的手册里 https://mpv.io/manual/master/#options-glsl-shader 应该讲的比较明白了
!DESC xxxx 这个对应的就是stats里每一个hook显示的名称
!WHEN xxx 限制了hook满足什么样的最低条件才能载入
至于其它的函数是什么意思和用法..我也不清楚，它默认你很了解opengl着色器的原理和语法所以手册这部分显得比较简单。
